src/coreclr/tools/aot/ILCompiler/Program.cs

```diff
@@ -80,7 +80,15 @@ public int Run()
             // The runtime expects certain baselines that the codegen can assume as well.
             if ((targetArchitecture == TargetArchitecture.X86) || (targetArchitecture == TargetArchitecture.X64))
             {
-                instructionSetSupportBuilder.AddSupportedInstructionSet("sse2"); // Lower baselines included by implication
+                if (targetOS == TargetOS.OSX)
+                {
+                    // The oldest hardware that is still supported by Apple as of macOS 10.15 is Intel Core i5-3330S.
+                    instructionSetSupportBuilder.AddSupportedInstructionSet("avx");
```

Comment:
but this bumps it to avx1 only

Reply:
I guess we can use `x86-x64-v3` here instead

Reply:
Correct. That's the baseline for macOS 10.15.

I don't think there was any official decision to update the OS minimum requirement to macOS 11. It would likely make sense by the time .NET 8 is released though. macOS 10.15 went out of support recently.

Reply:
`x86-x64-v3` implies AVX2 and we are not there yet unless the minimum macOS version is bumped to 11+.

Reply:
Ah I see what you mean, right 🙂